core/src/main/scala/cats/StackSafeMonad.scala

```diff
@@ -9,12 +9,38 @@ import scala.util.{Either, Left, Right}
  * will inherit will not be sound, and will result in unexpected stack overflows.  This
  * trait is only provided because a large number of monads ''do'' define a stack-safe
  * flatMap, and so this particular implementation was being repeated over and over again.
+ *
+ * Note, tailRecM being safe and pure implies that the function passed to flatMap
+ * is not called immediately on the current stack (since otherwise tailRecM would
+ * stack overflow for sufficiently deep recursions). This implies we can implement
+ *
+ * defer(fa) = unit.flatMap(_ => fa)
  */
-trait StackSafeMonad[F[_]] extends Monad[F] {
+trait StackSafeMonad[F[_]] extends Monad[F] with Defer[F] {
 
   override def tailRecM[A, B](a: A)(f: A => F[Either[A, B]]): F[B] =
     flatMap(f(a)) {
       case Left(a)  => tailRecM(a)(f)
       case Right(b) => pure(b)
     }
+
+  /**
```

Comment:
```suggestion
  /*
```

Can we actually make this *not* scaladoc? The problem here is if we make it a scaladoc comment, it will override the inherited one from `Defer`, which in turn results in *this* being the documentation shown in downstream generated scaladoc.